app/models/developers/external_profiles/linkedin_profile_fetcher.rb

```diff
@@ -0,0 +1,47 @@
+module Developers::ExternalProfiles
+  class LinkedinProfileFetcher
+    def developer_profiles
+      profiles = []
+      developers_with_linked_in_profiles.each do |developer|
+        response = fetch_linkedin_profile(developer.linkedin)
+        record = external_profile(developer, response)
+        profiles << record if record.present?
+      end
+
+      upsert_external_profiles(profiles)
+    end
+
+    def external_profile(developer, response)
+      if response[:error]
+        {developer_id: developer.id, site: "linkedin", data: {}, error: response[:error]}
+      else
+        external_profile = Developers::ExternalProfile.linkedin_developer(developer)
+        external_profile.data = response[:data] unless external_profile.blank?
+        if external_profile.blank? || external_profile.data_changed?
+          {developer_id: developer.id, site: "linkedin", data: response[:data], error: nil}
+        end
+      end
+    end
+
+    def upsert_external_profiles(profiles)
+      unless profiles.empty?
+        Developers::ExternalProfile.upsert_all(profiles, unique_by: [:developer_id, :site])
+      end
+    end
+
+    private
+
+    def developers_with_linked_in_profiles
+      Developer.where.not(linkedin: [nil, ""])
+    end
+
+    def fetch_linkedin_profile(linkedin_id)
+      linkedin_url = "https://linkedin.com/in/#{linkedin_id}/"
```

Comment:
It looks like some developers have trailing and/or leading slashes in their LinkedIn field. For example, `/foo-bar` or `/baz-qux/`.

Should we handle that here or normalize data when updating your profile + a task to clean existing data up?

Reply:
I think it will be better to handle while updating profile and running rake task to clean existing data as it not only effects this code but other places wherein we display LinkedIn profile details https://github.com/joemasilotti/railsdevs.com/blob/main/app/components/social_link_component.rb#L28

Reply:
Have added code to normalized LinkedIn username in existing file https://github.com/joemasilotti/railsdevs.com/blob/main/app/models/concerns/developers/has_online_profiles.rb

We already have rake task added for normalizing online public profiles which we can run again `rake backfills:url_attributes`

Reply:
Good call - thanks!